src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -104,7 +104,7 @@ private function completeDefinition($id, Definition $definition)
                 // Typehint against a non-existing class
 
                 if (!$parameter->isDefaultValueAvailable()) {
-                    continue;
+                    throw new RuntimeException(sprintf('Cannot auto-wire argument %s for %s because the type-hinted class does not exist.', $index + 1, $definition->getClass()));
```

Comment:
Why using 1-indexed numbers instead of 0 indexed? We do the same in other parts?